### PR TITLE
Proposal: compose run supports  quoted multi-argument commands

### PR DESCRIPTION
**Pull request by akirakoyasu**
Addresses Issue #148
reference #229

`compose run` supports multi-argument commands as:
`<container1> “<cmd0> <cmd1> <cmd2>” <container2> “<cmd0> <cmd1> <cmd2>...”`

### CODE REVIEW

File: ecs-cli/modules/aws/clients/ecs/client.go [hunk 1]
```diff
@@ -28,8 +28,8 @@ import (
 	"github.com/aws/aws-sdk-go/service/ecs/ecsiface"
 )
 
-//go:generate mockgen.sh github.com/aws/amazon-ecs-cli/ecs-cli/modules/aws/clients/ecs ECSClient mock/$GOFILE
```
Comment (yinshiua) [inline ecs-cli/modules/aws/clients/ecs/client.go]: This shouldn't change. Makefile adds this to the `PATH`

File: ecs-cli/modules/compose/cli/ecs/app/app.go [hunk 2]
```diff
@@ -88,12 +89,16 @@ func ProjectPs(p ecscompose.Project, c *cli.Context) {
 // ProjectRun starts containers and executes one-time command against the container
 func ProjectRun(p ecscompose.Project, c *cli.Context) {
 	args := c.Args()
-	if len(args)%2 != 0 {
+	if len(args) % 2 != 0 {
 		log.Fatal("Please pass arguments in the form: CONTAINER COMMAND [CONTAINER COMMAND]...")
```
Comment (yinshiua) [inline ecs-cli/modules/compose/cli/ecs/app/app.go]: Change the text to "Please pass arguments in the form: CONTAINER \"COMMAND ...\" [CONTAINER \"COMMAND...\"] ..."

File: ecs-cli/modules/compose/cli/ecs/app/app.go [hunk 3]
```diff
@@ -88,12 +89,16 @@ func ProjectPs(p ecscompose.Project, c *cli.Context) {
...
 	}
-	commandOverrides := make(map[string]string)
+	commandOverrides := make(map[string][]string)
 	for i := 0; i < len(args); i += 2 {
-		commandOverrides[args[i]] = args[i+1]
+		parts, err := shlex.Split(args[i + 1])
+		if err != nil {
+			log.Fatal(err)
```
Comment (yinshiua) [inline ecs-cli/modules/compose/cli/ecs/app/app.go]: To be more specific here.
```
log.WithFields(log.Fields{
    "error": err,
}).Fatal("Unable to parse run commands")
```

File: ecs-cli/modules/compose/ecs/project.go [hunk 4]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/docker/libcompose/project"
 )
 
+//go:generate ../../../../scripts/mockgen.sh github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/ecs Project mocks/$GOFILE
```
Comment (yinshiua) [inline ecs-cli/modules/compose/ecs/project.go]: Same here. Please revert this line

**Comment by akirakoyasu** [comment]
@yinshiua updated.

File: ecs-cli/modules/compose/cli/ecs/app/app.go [hunk 5]
```diff
@@ -88,12 +89,18 @@ func ProjectPs(p ecscompose.Project, c *cli.Context) {
...
+		log.Fatal("Please pass arguments in the form: CONTAINER \"COMMAND ...\" [CONTAINER \"COMMAND...\"] ...")
 	}
-	commandOverrides := make(map[string]string)
+	commandOverrides := make(map[string][]string)
 	for i := 0; i < len(args); i += 2 {
-		commandOverrides[args[i]] = args[i+1]
+		parts, err := shlex.Split(args[i + 1])
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error": err,
```
Comment (yinshiua) [inline ecs-cli/modules/compose/cli/ecs/app/app.go]: To help debugging, also add
```"container-name": arg[i]```

File: ecs-cli/modules/aws/clients/ecs/client.go [hunk 6]
```diff
@@ -399,13 +399,13 @@ func (c *ecsClient) RunTask(taskDefinition, startedBy string, count int) (*ecs.R
 }
 
 // RunTask issues a run task request for the input task definition
-func (c *ecsClient) RunTaskWithOverrides(taskDefinition, startedBy string, count int, overrides map[string]string) (*ecs.RunTaskOutput, error) {
+func (client *ecsClient) RunTaskWithOverrides(taskDefinition, startedBy string, count int, overrides map[string][]string) (*ecs.RunTaskOutput, error) {
```
Comment (yinshiua) [inline ecs-cli/modules/aws/clients/ecs/client.go]: go-lint error, need to change "client" to "c"

**Comment by akirakoyasu** [comment]
@yinshiua updated.

**Comment by yinshiua** [comment]
LGTM.
@akirakoyasu Are you ok contributing this code under Apache 2.0 license?

**Comment by akirakoyasu** [comment]
> @akirakoyasu Are you ok contributing this code under Apache 2.0 license?

@yinshiua Yes, ok.

**Pull request merged**